kahuna/public/js/edits/image-editor.js

```diff
@@ -73,6 +75,18 @@ imageEditor.controller('ImageEditorCtrl',
         ctrl.error = true;
     }
 
+    mediaApi.canDelete(ctrl.image).then(deletable => {
+        ctrl.canBeDeleted = deletable;
+    });
+
+
+    ctrl.onDeleteSuccess = () => {
+        $state.go('upload', {}, {reload: true});
```

Comment:
In this instance, I don't think a global event emission is required. Instead, the `gr-delete-image` shouldn't actually belong in `ui-image-editor` instead, it should be [here](https://github.com/guardian/grid/blob/master/kahuna/public/js/upload/view.html#L28). That way, the upload controller can remove the image from `this.myUploads` in the `onSuccess` argument of `gr-delete-image`.